### ISSUE TITLE
[BUG] google-gke autopilot cluster errors out when attempting to import to rancher

### ISSUE THREAD

**Issue by ronhorton** [rancher/rancher#41654]
**Rancher Server Setup**
- Rancher version: `v2.7-head commit ID 707d173`
- Installation option (Docker install/Helm Chart): `helm`
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): `rke2`
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: `1.25.8-gke.500`
- Cluster Type (Local/Downstream): `hosted GKE`
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions: `admin`

**Describe the bug**
created gke cluster with google autopilot; could not import cluster to rancher.
manually created cluster successfully imported. 
two attempts to import different auto-pilot cluster failed with same error.

**To Reproduce**
1. create gke auto-pilot cluster
2. import to rancher

**Result**
error - gke indicating that this object cannot have a particular key

**Expected Result**
cluster is imported 

**Screenshots**

<img width="1291" alt="2023-05-22_16-19-05" src="https://github.com/rancher/rancher/assets/89212856/f53fc820-d07d-40d6-9135-6b7072b69692">

**Additional context**

**Comment by ronhorton** [comment]
update: principal is not being created for the okta user

**Comment by cpinjani** [comment]
It is already [documented](https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/set-up-clusters-from-hosted-kubernetes-providers/gke#2-create-the-gke-cluster) while provisioning GKE cluster: "To successfully create a GKE cluster with Rancher, your GKE must be in Standard mode. GKE has two modes of operation when creating a Kubernetes cluster, Autopilot and Standard mode."

It would be worth documenting for imported cluster too in case no plan to support the same.

**Comment by davidstauffer** [comment]
Hi, as already pointed out from @cpinjani we don't support the provisioning of GKE clusters that are in autopilot mode. I quickly synced with @leflambeur and the best path forward is that we will start to support GKE autopilot imported, once we have added the enhancement/capability in general to rancher. 
Considering the above, we can probably park this issue for now, until we support Autopilot clusters.

**Comment by ornell** [comment]
> 

Hi, customer here and (probably one of the) originator of this issue in itself. Updating the docs after the fact to document that you have an issue does not solve the issue itself. Since autopilot now is the **default** way to provision clusters in GCP this is starting to become urgent if you would like co keep this market segment. (Take a moment to think about the fact that you do not support provisioning or importing GKE clusters in GCP with **default** settings.)

After doing some research here myself most of the issues that can be seen in the provisioning (Multiple layers after first problem has been resolved) is based on how you are doing node affinities with kubernetes.io/os. A full GKE import is probably not necessary at this point, but this is also blocks generic import as well. _The decision to wait on this is an active decision on not properly supporting GKE for the short term._

**Comment by richardcase** [comment]
This is a duplicate of #39052

**Comment by Jono-SUSE-Rancher** [comment]
Closing in favor of #39052
